server/src/main/java/io/spine/server/entity/Repository.java

```diff
@@ -371,6 +375,23 @@ protected void logError(String msgFormat,
         log().error(errorMessage, exception);
     }
 
+    /**
+     * Checks that {@link Column} definitions are correct for the {@link Entity} class
+     * managed by this repository.
+     *
+     * <p>This method relies on the {@linkplain Columns#checkColumnDefinitions(Class) Columns}
+     * utility to perform the check.
+     *
+     * <p>If {@link Column} definitions are incorrect, the {@link IllegalStateException}
+     * is thrown.
+     *
+     * @throws IllegalStateException in case entity column definitions are incorrect
+     */
+    @Internal
+    void checkEntityColumnDefinitions() {
```

Comment:
It seems to me, that a generic `Repository` cannot have the entity columns in all cases. It's `RecordBasedRepository`, that has. Therefore I think this method is too high in the `Repository` hierarchy.

Let's discuss.